core/trino-main/src/main/java/io/trino/metadata/AbstractCatalogPropertyManager.java

```diff
@@ -43,6 +44,25 @@ public void removeProperties(CatalogName catalogName)
         doRemoveProperties(catalogName);
     }
 
+    public Properties getProperties(
+            CatalogName catalog,
+            String catalogNameForDiagnostics,
+            List<Property> inputProperties,
+            Session session,
+            Metadata metadata,
+            AccessControl accessControl,
+            Map<NodeRef<Parameter>, Expression> parameters)
+    {
+        return doGetProperties(
+                catalog,
+                catalogNameForDiagnostics,
+                inputProperties,
+                session,
+                metadata,
+                accessControl,
+                parameters);
+    }
+
     public Map<String, Object> getProperties(
```

Comment:
We can just have one actual method here, e.g:
```
    public Properties getProperties(
            CatalogName catalog,
            String catalogNameForDiagnostics,
            List<Property> inputProperties,
            Session session,
            Metadata metadata,
            AccessControl accessControl,
            Map<NodeRef<Parameter>, Expression> parameters)
```
Then this method can be called `getNonNullProperties`:
```
Map<String, Object> getNonNullProperties(...) {
  return getProperties(...).getNonNullProperties();
}
```

Reply:
The difference between those two versions is more than just the return type (for example, one of them has a boolean parameter `setDefaultProperties` but the other does not). But anyway, I did some refactoring. Please take a look.